src/wagtail_ai/panels.py

```diff
@@ -43,3 +52,82 @@ class BoundPanel(AIPanelMixin.BoundPanel, FieldPanel.BoundPanel):  # type: ignor
 class AITitleFieldPanel(AIPanelMixin, TitleFieldPanel):
     class BoundPanel(AIPanelMixin.BoundPanel, TitleFieldPanel.BoundPanel):  # type: ignore
         pass
+
+
+class AIChooserPanelMixin(Panel):
+    def __init__(
+        self,
+        *args,
+        suggest_limit: int = 3,
+        chunk_size: int | None = None,
+        vector_index: "str | VectorIndex",
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+        self.suggest_limit = suggest_limit
+        self.chunk_size = chunk_size
+        if isinstance(vector_index, str):
+            self.vector_index = vector_index
+        else:
+            self.vector_index = vector_index.index_id
+
+    def clone_kwargs(self):
+        return {
+            **super().clone_kwargs(),
+            "vector_index": self.vector_index,
+            "suggest_limit": self.suggest_limit,
+            "chunk_size": self.chunk_size,
+        }
+
+    @classproperty
+    def BASE_ATTRS(cls):
+        base = super().BASE_ATTRS
+        controllers = [base.get("data-controller", ""), "wai-suggestions"]
+        actions = [
+            base.get("data-action", ""),
+            "w-formset:removed->wai-suggestions#updateControlStates",
+            "w-formset:added->wai-suggestions#updateControlStates",
+            "w-formset:ready->wai-suggestions#updateControlStates",
+        ]
+        return {
+            **base,
+            "data-controller": " ".join(controllers).strip(),
+            "data-action": " ".join(actions).strip(),
+        }
+
+    class BoundPanel(Panel.BoundPanel):
+        template_name = "wagtail_ai/panels/suggestions_panel.html"
+
+        @classproperty
+        def original_template_name(cls):
+            return super().template_name
+
+        @property
+        def attrs(self):
+            attrs = super().attrs
+            attrs["data-wai-suggestions-relation-name-value"] = self.panel.relation_name
+            attrs["data-wai-suggestions-instance-pk-value"] = (
+                self.instance.pk if self.instance else None
+            )
+            attrs["data-wai-suggestions-limit-value"] = self.panel.suggest_limit
+            attrs["data-wai-suggestions-vector-index-value"] = self.panel.vector_index
+            attrs["data-wai-suggestions-chunk-size-value"] = self.panel.chunk_size
+            return attrs
+
+        def get_context_data(self, parent_context=None):
+            context = super().get_context_data(parent_context=parent_context) or {}
+            context["parent_prefix"] = self.prefix.replace("content-child-", "content-")
```

Comment:
I think this might be a leftover, is it still used anywhere?

```suggestion
```

Reply:
Nope this was used in my code that shoved a reference to the panel on the DOM element, but your solution is much nicer so this can go.